lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
Also just to be clear - it is completely possible to do what you propose with this PR. It would only be relevant to the scenario no link is present.